ipatests/test_integration/test_trust.py

```diff
@@ -463,6 +357,50 @@ def test_user_uid_resolution_in_external_trust_with_shared_secret(self):
         assert re.search(
             testuser_regex, result.stdout_text), result.stdout_text
 
+    def test_override_homedir(self):
```

Comment:
You inserted the test case between steps of the test for external trust with shared secret, and at the same time you setup trust in your testcase, which is completely wrong. I see two options:
1) add this test case as a step of "Tests for posix AD trust". Do not re-establish and remove trust in this case
2) leave this test case as a self-contained scenario with trust setup and teardown but place it after all other tests.